sunbird-es-utils/src/main/java/org/sunbird/common/ElasticSearchTcpImpl.java

```diff
@@ -0,0 +1,652 @@
+import org.sunbird.common.models.util.PropertiesCache;
+import org.sunbird.common.responsecode.ResponseCode;
+import org.sunbird.dto.SearchDTO;
+import org.sunbird.helper.ConnectionManager;
+import scala.concurrent.Future;
+import scala.concurrent.Promise;
+
+/**
+ * This class will provide all required operation for elastic search.
+ *
+ * @author github.com/iostream04
+ */
+public class ElasticSearchTcpImpl implements ElasticSearchService {
+  public static final int WAIT_TIME = 30;
+  public static Timeout timeout = new Timeout(WAIT_TIME, TimeUnit.SECONDS);
+  private static final String _DOC = "_doc";
+  /**
+   * This method will put a new data entry inside Elastic search. identifier value becomes _id
+   * inside ES, so every time provide a unique value while saving it.
+   *
+   * @param index String ES index name
+   * @param identifier ES column identifier as an String
+   * @param data Map<String,Object>
+   * @return identifier for created data
+   */
+  @Override
+  public Future<String> save(String index, String identifier, Map<String, Object> data) {
+    long startTime = System.currentTimeMillis();
+    Promise<String> promise = Futures.promise();
+    ProjectLogger.log(
+        "ElasticSearchTcpImpl:save: method started at ==" + startTime + " for Index " + index,
+        LoggerEnum.PERF_LOG);
+    if (StringUtils.isBlank(identifier) || StringUtils.isBlank(index)) {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: Identifier value is null or empty ,not able to save data.",
+          LoggerEnum.ERROR);
+      promise.success("ERROR");
+      return promise.future();
+    }
+    try {
+      data.put("identifier", identifier);
+      IndexResponse response =
+          ConnectionManager.getClient().prepareIndex(index, _DOC, identifier).setSource(data).get();
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: "
+              + "Save value=="
+              + response.getId()
+              + " "
+              + response.status(),
+          LoggerEnum.INFO.name());
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: method end at =="
+              + System.currentTimeMillis()
+              + " for INdex "
+              + index
+              + " ,Total time elapsed = "
+              + ElasticSearchHelper.calculateEndTime(startTime),
+          LoggerEnum.PERF_LOG);
+      promise.success(response.getId());
+      return promise.future();
+    } catch (Exception e) {
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: Error while saving index " + index + " id : " + identifier,
+          e);
+      ProjectLogger.log(
+          "ElasticSearchTcpImpl:save: method end at =="
+              + System.currentTimeMillis()
+              + " for Index "
+              + index
+              + " ,Total time elapsed = "
+              + ElasticSearchHelper.calculateEndTime(startTime),
+          LoggerEnum.PERF_LOG);
+      promise.failure(e);
```

Comment:
why are are passing failure and success both

Reply:
We will passing the empty string in success as to accommodate consumers which are using it.